### PR TITLE
VIH-9682 [D01] Code Improvments

### PR DESCRIPTION
**Pull request by benjamin-v1**
### JIRA ###

[VIH-9682](https://tools.hmcts.net/jira/browse/VIH-9682)

### Change description ###

Terraform refactoring 

**Does this PR introduce a breaking change?** (check one with "x")

```
[X] Yes
[] No
```

### CODE REVIEW

**Comment by chris-pateman** [comment]
Can you create a ticket and update the title please

File: terraform/_versions.tf [hunk 1]
```diff
@@ -1,4 +1,6 @@
 terraform {
+  backend "azurerm" {
+  }
   required_version = ">= 1.0.4"
```
Comment (chris-pateman) [inline terraform/_versions.tf:4]: should we upgrade to 1.3.7?

File: terraform/envs/sbox.tfvars [hunk 2]
```diff
@@ -1,5 +1,6 @@
 environment                   = "sbox"
 env                           = "sbox"
+vm_count                      = 2
```
Comment (chris-pateman) [inline terraform/envs/sbox.tfvars:3]: Can we change this to 1?

File: terraform/modules/wowza/providers.tf [hunk 3]
```diff
@@ -0,0 +1,15 @@
+provider "azurerm" {
```
[the remark this answers is not in the document]
Comment (benjamin-v1) [reply terraform/modules/wowza/providers.tf:1]: This is required to allow the existing TF code to initiate. We can remove it once we have gone to prod once.

File: terraform/_versions.tf (hunk 1 above)
Comment (benjamin-v1) [reply terraform/_versions.tf:4]: yes, wanted to get this done first and stable, then go back and do a version bump. Happy to just do it at the same time if you think

File: terraform/modules/wowza/providers.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (chris-pateman) [reply terraform/modules/wowza/providers.tf:1]: perfect

File: terraform/envs/sbox.tfvars (hunk 2 above)
Comment (chris-pateman) [reply terraform/envs/sbox.tfvars:3]: perfect

File: terraform/_locals.tf [hunk 4]
```diff
@@ -0,0 +1,8 @@
+locals {
+  service_name              = "${var.product}-recordings-${var.env}"
+  domain_dns_prefix         = var.env == "stg" ? "aat" : var.env == "sbox" ? "sandbox" : var.env == "stg" ? "staging" : var.env
+  common_tags               = module.ctags.common_tags
```
[the remark this answers is not in the document]
Comment (benjamin-v1) [reply terraform/_locals.tf]: done

File: terraform/envs/sbox.tfvars (hunk 2 above)
Comment (benjamin-v1) [reply terraform/envs/sbox.tfvars]: done

File: terraform/virtual-machine-cloudconfig.tf [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    containerName           = local.main_container_name
+    logsContainerName       = local.wowza_logs_container_name
+    numApplications         = var.num_applications
+    managedIdentityClientId = azurerm_user_assigned_identity.mi.client_id
+    certName                = "cvp-${var.env}-le-cert"
+    keyVaultName            = "cvp-${var.env}-kv"
+    domain                  = "cvp-recording.${local.domain_dns_prefix}.platform.hmcts.net"
+    wowzaVersion            = var.wowza_version
+    dynatrace_tenant        = var.dynatrace_tenant
+    dynatrace_token         = var.env == "stg" || var.env == "prod" ? data.azurerm_key_vault_secret.dynatrace_token[0].value : ""
```
[the remark this answers is not in the document]
Comment (benjamin-v1) [reply terraform/virtual-machine-cloudconfig.tf]: done

**Review by chris-pateman — APPROVED** [review]
LGTM

File: pipeline/stages/test.yaml [hunk 6]
```diff
@@ -32,22 +32,52 @@ stages:
           displayName: '${{parameters.env}} Virtual Machine Tests '
           steps:
 
+            - task: AzureCLI@2
```
Comment (chris-pateman) [inline pipeline/stages/test.yaml:35]: should this be a template instead either local or remote?

File: terraform/README.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
+<!-- BEGIN_TF_DOCS -->
```
Comment (chris-pateman) [inline terraform/README.md:1]: maybe want to add info on updating the wowza version and the vm count if required?

**Review by chris-pateman — CHANGES_REQUESTED** [review]

File: pipeline/stages/test.yaml (hunk 6 above)
Comment (benjamin-v1) [reply pipeline/stages/test.yaml]: moving to template ...

File: terraform/README.md (hunk 7 above)
Comment (benjamin-v1) [reply terraform/README.md:1]: will overhaul docs once we have made other changes

**Review by chris-pateman — APPROVED** [review]
LGTM

**Pull request merged**